### PR TITLE
Enumerables branch

### PR DESCRIPTION
**Pull request by aracelicaes**
Pull request #1 for my enumerables-branch

### CODE REVIEW

File: ara_enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
```
Comment (Forison) [inline ara_enumerables.rb]: #### [optional]
Pass the method name as an argument to ```to_enum``` like you have done for my_each to make your method more customized.
``` return to_enum(:my_each_with_index) unless block_given?```

File: ara_enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    if block_given?
+      arr = []
+      my_each do |i|
+        arr << i if yield(i)
+      end
+    end
+    arr
+  end
+
+  def my_all?(arg = nil)
```
Comment (Forison) [inline ara_enumerables.rb:38]: * [x] should have working my_all? identical and returns the same thing as ruby's all?.  ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
let us fix this
```p [1, 2i, 3.14].my_all?(Numeric)  != p [1, 2i, 3.14].all?(Numeric)```

File: ara_enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    i = 0
+    while i < size
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
```
Comment (Forison) [inline ara_enumerables.rb]: #### [optional]
Pass the method name as an argument to ```to_enum``` like you have done for my_each to make your method more customized.
``` return to_enum(:my_select) unless block_given?```

**Review by Forison — CHANGES_REQUESTED** [review]
## Status: Require changes🛑
# 🙋‍♂️ Ara,
You have done a good job on this one. There is an issue with ``my_all``, let us fix it.
### check below the comments 👇👇👇 for the areas you have to improve.
## That is it kindly make the change and submit the project for review🙂
#### Happy coding💻
#### Addo Forison

File: utils.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+def add_it(initial)
+  sum = initial
+  my_each { |e| sum += e }
+  sum
+end
```
Comment (bolah2009) [inline utils.rb:5]: This method seems to be adding the first value of the array two times when no initial value is given in the `inject` method.

```ruby
irb(main):002:0> [1,2,3,4,5,56,7,8,9,234].inject(:+)
#=> 329
irb(main):003:0> [1,2,3,4,5,56,7,8,9,234].my_inject(:+)
#=> 330

irb(main):005:0> [19,2,3,4,5,56,7,8,9,234].inject(:+)
#=> 347
irb(main):006:0> [19,2,3,4,5,56,7,8,9,234].my_inject(:+)
#=> 366
```

**Review by bolah2009 — APPROVED** [review]
It's a 👍🏾 from me!

### :heavy_check_mark: Approved!

Good job implementing changes suggested by @Forison 

Check out the optional comments before merging. :relaxed: 

Good luck with your next projects. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Pull request merged**